src/browser/modules/D3Visualization/components/ResultsPane.tsx

```diff
@@ -137,49 +146,63 @@ export class LegendComponent extends Component<any, State> {
         return (
           <StyledLegendInlineListItem key={i} data-testid="viz-legend-reltypes">
             <StyledLegendContents className="contents">
-              <StyledTokenRelationshipType
-                onClick={onClick}
-                style={style}
-                className="token token-relationship-type"
+              <Popup
+                on="click"
+                basic
+                pinned
+                trigger={
+                  <StyledTokenRelationshipType
+                    onClick={onClick}
+                    style={style}
+                    className="token token-relationship-type"
+                  >
+                    {legendItemKey}
+                    <StyledTokenCount className="count">
+                      {`(${numberToUSLocale(
+                        legendItems[legendItemKey].count
+                      )})`}
+                    </StyledTokenCount>
+                  </StyledTokenRelationshipType>
+                }
+                wide
+                inverted={this.props.theme === DARK_THEME}
               >
-                {legendItemKey}
-                <StyledTokenCount className="count">
-                  {`(${numberToUSLocale(legendItems[legendItemKey].count)})`}
-                </StyledTokenCount>
-              </StyledTokenRelationshipType>
+                <GrassEditor
+                  selectedRelType={
+                    this.props.selectedLabel?.item?.selectedRelType
+                  }
+                  frameHeight={this.props.frameHeight}
+                />
+              </Popup>
             </StyledLegendContents>
           </StyledLegendInlineListItem>
         )
       })
       return (
-        <StyledLegendRow
-          className={this.state.typeRowContracted ? 'contracted' : ''}
-        >
+        <StyledLegendRow>
           <StyledLegendInlineList
             className="list-inline"
             ref={this.setTypeRowELem.bind(this)}
           >
-            <RowExpandToggleComponent
-              contracted={this.state.typeRowContracted}
-              rowElem={this.typeRowElem}
-              containerHeight={legendRowHeight}
-              onClick={() => {
-                this.setState({
-                  typeRowContracted: !this.state.typeRowContracted
-                })
-              }}
-            />
             {relTypeList}
           </StyledLegendInlineList>
         </StyledLegendRow>
       )
     }
     const relTypes = mapRelTypes(this.props.stats.relTypes)
     return (
-      <StyledLegend className={relTypes ? '' : 'one-row'}>
-        {mapLabels(this.props.stats.labels)}
-        {relTypes}
+      <StyledLegend>
+        Node Labels:{' '}
+        {mapLabels(this.props.stats.labels) || <div>No labels to display</div>}
+        Relationship Types:{' '}
+        {relTypes || <div>No relationship types to display</div>}
```

Comment:
unstyled fallback text as explained in PR description - its open to being handled in multiple ways so these are just placeholders for now